Sources/Prometheus/MetricTypes/Histogram.swift

```diff
@@ -233,14 +233,23 @@ public class PromHistogram<NumType: DoubleRepresentable, Labels: HistogramLabels
 
     /// Helper for histograms & labels
     fileprivate func getOrCreateHistogram(with labels: Labels) -> PromHistogram<NumType, Labels> {
-        let histogram = self.subHistograms[labels]
-        if let histogram = histogram, histogram.name == self.name, histogram.help == self.help {
-            return histogram
+        let subHistograms = lock.withLock { self.subHistograms }
+        if let histogram = subHistograms[labels] {
+            if histogram.name == self.name, histogram.help == self.help {
+                return histogram
+            } else {
+                fatalError("Somehow got 2 summaries with the same data type")
+            }
         } else {
-            guard let prometheus = prometheus else { fatalError("Lingering Histogram") }
-            let newHistogram = PromHistogram(self.name, self.help, labels, Buckets(self.upperBounds), prometheus)
-            self.subHistograms[labels] = newHistogram
-            return newHistogram
+            return lock.withLock {
+                if let histogram = subHistograms[labels], histogram.name == self.name, histogram.help == self.help {
```

Comment:
Above we `fatalError` if name and help don't match up. Do we want to do the same here?